pkg/registry/core/pod/strategy.go

```diff
@@ -281,6 +268,65 @@ func (podEphemeralContainersStrategy) WarningsOnUpdate(ctx context.Context, obj,
 	return nil
 }
 
+type podResizeStrategy struct {
+	podStrategy
+}
+
+// ResizeStrategy wraps and exports the used podStrategy for the storage package.
+var ResizeStrategy = podResizeStrategy{Strategy}
```

Comment:
this has to implement GetResetFields() and enumerate *every* field path that gets reset for server-side apply to work properly

```
spec.... (every individual parent path to fields that are untouched by requests to this endpoint)
metadata
status
```

Reply:
Seems like more reason to not reset the spec fields, and instead fail it in validation?

Reply:
I checked and it looks like /ephemeralcontainers fails validation.  I had assumed it reset fields, but I was wrong.

Maybe that consistency with /ephemeralcontainers + problems with GetResetFields() is enough reason to validate here?

I'll defer to @thockin and @liggitt for API review decision here.

Reply:
> I am not super versed in how SSA actually works, but it seems like enumerating every field is going to be a nightmare. Can't we somehow say "all except..." ?

I've opened https://github.com/kubernetes/kubernetes/issues/128345 to track this, since it seems like something we need to address.  If the outcome of the "should we passing a whole pod type here" ends up being a *yes*, please ping the issue and SIG API Machinery can bump the prioritization.

Reply:
Done.